### ISSUE TITLE
acSet not defined

### ISSUE THREAD

**Issue by rt-2** [NexiusTailer/Nex-AC#22]
There is no documentation about acSet, it is an array of 15, but I cannot know what is inside.
I need to work with [0]. [12] and [14] because they are part of the anti-interior hack, and the anti-interior hack is bugged.
But I would like to know what I am working with.

What is all the information stored in acSet please?
Thank you,

**Comment by NexiusTailer** [comment]
The acSet array stores the values of various functions that the server only wants to apply for player (ie the function has already done, but the player has not yet received new data). Those it is something like an intermediate storage of this information.

ACInfo[playerid][acSet][0] stores the value of the interior (if the function SetPlayerInterior was used before) and -1, if at the moment the player does not wait for receive a new interior.
But if the player at the time of setting the interior is dead, then the interior will be written in acSet[12] (because in OnplayerSpawn acSet[0] will be assigned to -1, and the player's interior will automatically change to 0, and without an additional reserve variable it might create false detection at spawn).
Then in OnPlayerInteriorChange, if the [[first check](https://github.com/NexiusTailer/Nex-AC/blob/master/src/v1.9/v1.9.22/r2/nex-ac.inc#L3224)] fails (ie will executed), [[the second one](https://github.com/NexiusTailer/Nex-AC/blob/master/src/v1.9/v1.9.22/r2/nex-ac.inc#L3242)] will be executed, and only after its failure the player will be punished.
So this is the main reason why there is a variable ACInfo[playerid][acSet][12].

With the variable acSet[14], everything is much simpler: it is needed to protect against setting several interiors through SetPlayerInterior at the same time. I do not remember exactly whether it is needed in this version (because at the time of adding this variable there were some problems with setting different interiors at once, which are now fixed).

**Comment by NexiusTailer** [comment]
P.s. Can you describe more detail the bug, with which you're facing?

**Comment by rt-2** [comment]
Thank you for your answer, the bug was on my part of the script.
Thank you for the information that helped me debug,
Sorry for not updating.

Thank you, talk to you later

**Comment by rt-2** [comment]
I need ALL informations on acSet, I wanna know what they ALL mean....
This SHOULD be better documented, I am trying to find out more about issue #46 but the variable names don't make any sense so it's impossible.
Thank you.

**Comment by WheneverRide** [comment]
acSet - is array of what server wants to do with player.
0 - set interior ID.
1 - set health.
2 - set armour.
3 - set weapon in arms (select weapon).
4 - set special action ID.
5 - set seat place in vehicle.
6 - put player in spectate mode.
7 - what reason forced player to spawn (-1 OnPlayerSpawn, 1 - exiting from spec, 2 - SpawnPlayer, 3 - OnPlayerSpawnRequest).
8 - Fact of player or it's vehicle teleportation by the server.
9 - vehicle ID from PutPlayerInVehicle.
10 - summ of money that must be wasted by player due different reasons.
11 - ejection player from vehicle fact.
